test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java

```diff
@@ -0,0 +1,220 @@
+    String metadataTable = tableNames[0];
+
+    var tabletTime = TabletTime.getInstance(new MetadataTime(1, TimeType.LOGICAL));
+    long flushNonce = 42L;
+
+    try (var client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(metadataTable);
+      TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      Ample testAmple = TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      assertNull(testAmple.readTablet(extent));
+
+      // create tablet in the fake metadata table
+      testAmple.mutateTablet(extent).putDirName("dir1").putTime(tabletTime.getMetadataTime())
+          .putLocation(Location.current(tserverInstance)).putPrevEndRow(null).mutate();
+
+      // get the tablets metadata
+      var lastMetadata = testAmple.readTablet(extent);
+
+      // update the tablets time, so it will differ
+      testAmple.mutateTablet(extent).putTime(tabletTime.getMetadataTime(tabletTime.getTime() + 2))
+          .mutate();
+
+      tabletTime.updateTimeIfGreater(tabletTime.getTime() + 6);
+
+      updateTabletDataFile(testAmple, tabletTime.getTime(), newFile, dfv1, Set.of(), 7L,
+          MinorCompactionReason.SYSTEM, tserverInstance, extent, lastMetadata, tabletTime,
+          flushNonce);
+
+      var updatedMetadata = testAmple.readTablet(extent);
+      assertEquals(Set.of(newFile.insert()), updatedMetadata.getFiles());
+      assertEquals(tabletTime.getMetadataTime(), updatedMetadata.getTime());
+      assertEquals(flushNonce, updatedMetadata.getFlushNonce().orElse(-1));
+
+      client.tableOperations().delete(metadataTable);
+    }
+  }
+
+  /**
+   * This test ensures that a tablet will not add a new minor compacted file when the tablets
+   * location is not as expected.
+   */
+  @Test
+  public void testLocation() throws Exception {
+    String[] tableNames = getUniqueNames(1);
+    String metadataTable = tableNames[0];
+
+    var tabletTime = TabletTime.getInstance(new MetadataTime(1, TimeType.LOGICAL));
+    long flushNonce = 42L;
+
+    try (var client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(metadataTable);
+      TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      Ample testAmple = TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      assertNull(testAmple.readTablet(extent));
+
+      // create tablet in the fake metadata table
+      testAmple.mutateTablet(extent).putDirName("dir1").putTime(tabletTime.getMetadataTime())
+          .putLocation(Location.current(tserverInstance)).putPrevEndRow(null).mutate();
+
+      // get the tablets metadata
+      var lastMetadata = testAmple.readTablet(extent);
+
+      // TODO open an issue about making requireLocation more strict
```

Comment:
Likely need to resolve this before merging

Reply:
Removed the todo and opened #4620.  While working on this I wrote a buggy version of this test that did not fail in the way I expected and discovered #4620 in the process.  After discovering the test bug I fixed the test and added the todo.